### PR TITLE
Implement project list page component

### LINKED ISSUE #1341: Rework projects list page and the ManageProjects component
### Overview
The current `ManageProjects` component contains too much functionality. For this issue, we will create a new `ProjectList` component to replace the `ManageProjects` component for the `/projects` route.

### Action Items
- [x] Create a new file in the `pages/` folder called `ProjectList.js`
- [x] Implement the new Project List page design in the new `ProjectList` component by completing the following
    - [x]  Make an API call to the backend route that responds with a list of all the projects and keep that data in state
    - [x] Loop over the projects data and display each project on the page in a list format as shown in the figma design
    - [x] Add the new project button that leads to the form to create a new project
    - [x] Match all the styles with the figma provided in the resources below
- [x] Test the `/projects/:projectId` route to make sure it still works

### Resources/Instructions
- https://www.figma.com/file/pIxFTL62Nx2XyQdZy1fR6f/VRMS-v.4?node-id=1093-1155&t=iYpURHLtqW4g2c3B-0

<details><summary>See screenshot of new project list page design</summary>

image

</details>

### PR DESCRIPTION
**Pull request by MattPereira**
Fixes #1341

### What changes did you make and why did you make them ?
  - Created a new page component `ProjectList` that is displayed when user is routed to `/projects` because the `ManageProjects` component is unorganized and prone to bugs
  - Left the `ManageProjects` page component untouched for now but will try to refactor with a future issue. This may also help narrow down where to search for the sever not responding bug #1321
- Set up h3 styles in `theme.js` for the headers inside of the grey boxes
- Added default font `Source Sans Pro` to `theme.js`
- Admin can see all projects and the button for adding a new project
- Project managers can only see projects they manage and will not see new project button

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image

</details>

### CODE REVIEW

File: client/src/pages/ProjectList.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          variant="secondary"
+          sx={{ mb: 3 }}
+        >
+          Add a New Project
+        </Button>
+      </Box>
+
+      <Box sx={{ bgcolor: '#F5F5F5' }}>
+        <Box sx={{ p: 2 }}>
+          <Typography sx={{ fontFamily: 'Source Sans Pro', fontWeight: 600 }}>
```
Comment (trillium) [inline client/src/pages/ProjectList.js]: suggest variant="h2" or "h3"

File: client/src/pages/ProjectList.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          </Typography>
+        </Box>
+        <Divider sx={{ borderColor: 'rgba(0,0,0,1)' }} />
+        <Box sx={{ p: 2 }}>
+          {projects.map((project) => (
+            <Box key={project._id} sx={{ mb: 0.5 }}>
+              <Typography
+                sx={{
+                  fontFamily: 'Source Sans Pro',
+                  textTransform: 'uppercase',
```
Comment (trillium) [inline client/src/pages/ProjectList.js]: Suggest a hover or focus state so that the user has feedback that these items are links

**Review by trillium — APPROVED** [review]
Good, clean, nice comments. Suggested two changes, those aren't blockers for me though.

File: client/src/pages/ProjectList.js (hunk 1 above)
Comment (MattPereira) [reply client/src/pages/ProjectList.js]: Good call i was just being lazy not setting that up yet but will do now :+1:

File: client/src/pages/ProjectList.js (hunk 2 above)
Comment (MattPereira) [reply client/src/pages/ProjectList.js]: Love this idea as well! consider it implemented

**Pull request merged**